src/krux/firmware.py

```diff
@@ -258,17 +259,19 @@ def status_text(text):
         display.clear()
         display.draw_centered_text(text)
 
+    firmware_file = open(firmware_path, "rb", buffering=0)
     write_data(
         lambda pct: status_text(
             t("Processing..") + "1/3" + "\n\n%d%%" % int(pct * 100)
         ),
         new_address,
-        open(firmware_path, "rb", buffering=0),
+        firmware_file,
         new_size,
         65536,
         True,
         firmware_with_header_hash,
     )
+    firmware_file.close()
```

Comment:
Might this need to be within the "finally" branch of try/except, in case of a write failure? or within the context of "with open()" so that it's automagically closed after the write_data() call?

Reply:
Will change that!